modules/dev-tools/deprecation.php

```diff
@@ -199,25 +199,189 @@ private function check_deprecation( $entity, $version, $replacement, $base_versi
 			throw new \Exception( 'Invalid deprecation diff.' );
 		}
 
-		$print_deprecated = false;
+		$print_deprecated = array();
 
-		if ( defined( 'WP_DEBUG' ) && WP_DEBUG && $diff <= self::SOFT_VERSIONS_COUNT ) {
-			// Soft deprecated.
-			if ( ! isset( $this->soft_deprecated_notices[ $entity ] ) ) {
-				$this->soft_deprecated_notices[ $entity ] = [
-					$version,
-					$replacement,
-				];
+		if ( $this->should_print_deprecated( $version, $base_version ) || $this->should_console_log_deprecated() ) {
+			$backtrace = debug_backtrace( DEBUG_BACKTRACE_IGNORE_ARGS );
+
+			$external_sources = $this->find_third_party_sources( $backtrace );
+
+			$source_message = 'Elementor';
+			$plugin_name = 'unknown';
+			if ( ! ( empty( $external_sources ) ) ) {
+				$calling_source = $external_sources[ array_key_first( $external_sources ) ];
+				$plugin_name = $this->get_plugin_name( $calling_source['file'] );
```

Comment:
Maybe move to `find_third_party_sources`

Reply:
I'd rather not, prefer to keep find_third_party_sources() as pure and designated as possible